tests/e2e/pages/vscode.page.ts

```diff
@@ -181,6 +183,140 @@ export class VSCode extends BasePage {
+   * @param order - The sort order ('ascending' or 'descending').
+   */
+  public async setListKindAndSort(kind: ListKind, order: SortOrder): Promise<void> {
+    const analysisView = await this.getView(KAIViews.analysisView);
+    const kindButton = analysisView.getByRole('button', {
+      name: kind === 'issues' ? 'Issues' : 'Files',
+    });
+    await kindButton.click();
+    await expect(kindButton).toHaveAttribute('aria-pressed', 'true');
+    const sortButton = analysisView.getByRole('button', {
+      name: order === 'ascending' ? 'Sort ascending' : 'Sort descending',
+    });
+    await sortButton.click();
+    await expect(sortButton).toHaveAttribute('aria-pressed', 'true');
+    await this.window.waitForTimeout(2000);
+  }
+
+  /**
+   * Retrieves the names of items (issues or files) currently listed in the analysis view.
+   * @param _ - The kind of list ('issues' or 'files').
+   * @returns A promise that resolves to an array of item names.
+   */
+  async getListNames(_: ListKind): Promise<string[]> {
+    const view = await this.getView(KAIViews.analysisView);
+    const listCards = view
+      .locator('[data-ouia-component-type="PF6/Card"]')
+      .filter({ has: view.getByRole('heading', { level: 3 }) })
+      .filter({ has: view.getByRole('button', { name: /get solution/i }) });
+    const titles = listCards.getByRole('heading', { level: 3 });
+    await expect(titles.first()).toBeVisible({ timeout: 6_000 });
+
+    const texts = await titles.allInnerTexts();
+    return texts.map((t) => t.replace(/\s+/g, ' ').trim()).filter(Boolean);
+  }
+
+  /**
+   * Opens the category filter dropdown in the analysis view and returns its elements.
+   * @returns An object containing the category button, menu, and options locators.
+   */
+  public async openCategory() {
+    const view = await this.getView(KAIViews.analysisView);
+    const categoryBtn = view.getByRole('button', { name: /^Category(?:\s*\d+)?$/i });
+    await categoryBtn.scrollIntoViewIfNeeded();
+
+    for (let i = 0; i < 3; i++) {
+      await categoryBtn.click();
+      try {
+        await expect(categoryBtn).toHaveAttribute('aria-expanded', 'true', { timeout: 1500 });
+        break;
+      } catch {
+        if (i === 2) throw new Error('Category button did not expand');
+      }
+    }
+
+    const controlsId = await categoryBtn.getAttribute('aria-controls');
+    let menu = controlsId ? view.locator(`#${controlsId}`) : null;
+
+    let options = view.locator(
+      '[role="menuitem"], [role="menuitemcheckbox"], [role="menuitemradio"], [role="option"]'
+    );
+    if (menu) {
+      options = (menu as any).locator(
+        '[role="menuitem"], [role="menuitemcheckbox"], [role="menuitemradio"], [role="option"]'
+      );
+    }
+
+    await expect
+      .poll(async () => (await options.count()) > 0 && (await options.first().isVisible()), {
+        timeout: 6_000,
+      })
+      .toBeTruthy();
```

Comment:
why do we need to poll here? isn't it the same to just call `options.waitFor()`?